include/circt/Dialect/OM/OMAttributes.td

```diff
@@ -59,4 +59,22 @@ def OMSymbolRefAttr : AttrDef<OMDialect, "SymbolRef", [TypedAttrInterface]> {
   }];
 }
 
+def OMStringAttr : AttrDef<OMDialect, "OMString", [TypedAttrInterface]> {
+  let summary = "An attribute that wraps a StringAttr with !om.string type";
```

Comment:
A tricky issue to wrap `StringAttr` with `OMStringAttr` is it makes hard to use common `var_to_attribute/attribute_to_var` python utility as we have to convert python `str` into `OMStringAttr`.

Reply:
Hmm this is annoying. I think we can probably add support for the OM dialect attributes to those helpers.